framework/doc/content/source/base/MooseObject.md

```diff
@@ -0,0 +1,18 @@
+# MooseObject.C
+
+The `MooseObject` is the top level base class for all non-action MOOSE objects
+(such as Kernels, Materials, Postprocessors, etc.).
+
+This class holds man convenience functions, such as the `getParam<>` template.
+
+## Data files (`getDataFileName`)
+
+Files located in `moose/framework/data`, the `moose/modules/*/data`, or `app/data` directories
+can be retrieved using the `getDataFileName(const std::string & param)` function, where
+`param` is an input parameter of type `FileName`
+
+`getDataFileName` will search (in this order)
+
+- relative to the input file
+- relative to the running binary in the shared directory (assuming the application is installed)
```

Comment:
What does it mean to be installed? I know that has been worked on recently...but practically does that change the location? just register some things? does it have to be run every time after make?

Reply:
Also, I assume the search also includes `test` data files, e.g. the test 316 we use in moose?

Reply:
It includes all paths relative to those search directories